### DISCUSSION TITLE
Max pool connection and BulkInsert

### DISCUSSION THREAD

**Discussion by gexcode** [sijms/go-ora#402] [Q&A] [answered]
I have an application running multiples threads to consume messages from a RabbitMQ queue and using the BulkInsert method to insert 1000 rows in Oracle database.
The DBA team requested to limit the open database connection to 50 (max).
The sql.DB allows to use the method SetMaxOpenConns, Is it possible to set the max pool size using go_ora and bulkInsert?

**Comment by sijms** [comment, marked as answer]
BulkInsert now can run from inside sql.Exec() just pass an arrays. you can review example merge/main.go
this is a code example
```golang
// all arrays should be same size
_, err := conn.Exec(sqlText,  // sql text INSERT INTO OR MERGE INTO
                sql.Named("ID", ids),   // array of integers
		sql.Named("NAME", names), // array of string
		sql.Named("VAL", vals),  // array of float
		sql.Named("LDATE", dates)) // array of time.Time
```
